internal/decoder/api/stream.go

```diff
@@ -76,11 +76,12 @@ func (self *StreamDecoder) Decode(val interface{}) (err error) {
         if y := native.SkipOneFast(&src, &x); y < 0 {
             if self.readMore()  {
                 goto try_skip
-            } else {
-                err = SyntaxError{e, self.s, types.ParsingError(-s), ""}
-                self.setErr(err)
-                return
+            }                
+            if self.err == nil {
```

Comment:
```
if self.err == nil {
    self.setErr(SyntaxError{e, self.s, types.ParsingError(-s), ""})
}
return self.err
```

no need to call `setErr` again if `self.err != nil `

Reply:
ok, fix